srml/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -0,0 +1,395 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+mod parse;
+
+use parse::{ModuleDeclaration, ModulePart, RuntimeDefinition, WhereSection};
+use proc_macro::TokenStream;
```

Comment:
Please use only `proc_macro2` down here.

Reply:
It will be cumbersome since `syn::parse_macro_input` works only with `proc_macro`. But I reduced the usage to the minimum